src/System.Xml.XmlSerializer/src/System/Xml/Serialization/CodeIdentifiers.cs

```diff
@@ -20,20 +20,19 @@ namespace System.Xml.Serialization
     /// </devdoc>
     internal class CodeIdentifiers
     {
-        private Hashtable _identifiers;
-        private Hashtable _reservedIdentifiers;
+        private Dictionary<string, object> _identifiers;
+        private Dictionary<string, object> _reservedIdentifiers;
         private ArrayList _list;
```

Comment:
At first I just removed the ```ArrayList``` shim type and use ```System.Collections.ArrayList``` instead. But it makes sense to update ```ArrayList``` to ```List<>``` as well. The ```System.Collections.NonGeneric``` would be unnecessary after that.